libs/core/src/components/pds-select/pds-select.tsx

```diff
@@ -0,0 +1,207 @@
+        }
+      });
+    };
+  }
+
+  /**
+   * Emits an event on input change
+  */
+  private onSelectUpdate = (e: Event) => {
+    const target = e.target as HTMLSelectElement
+
+    const values = Array.from(target.options)
+        .filter((option) => ( option.selected))
+        .map((option) => ( option.value));
+
+    if (values.length === 1 && !this.multiple) {
+        this.value = values[0];
+    } else {
+        this.value = values;
+    }
+
+    this.pdsSelectChange.emit(e as InputEvent);
+  };
+
+  /**
+   * Handles the change event for the slot element.
+   * This method is triggered when the slot content changes.
+   * It updates the inner HTML of the select element by cloning and appending
+   * the assigned <option> elements from the slot.
+   */
+  private handleSlotChange = () => {
+    const slot = this.slotContainer.querySelector('slot') as HTMLSlotElement;
+
+    this.selectEl.innerHTML = '';
+    const assignedElements = slot.assignedElements({ flatten: true }) as (HTMLOptionElement | HTMLOptGroupElement)[];
+
+    assignedElements.forEach((item) => {
+      if (item.tagName === 'OPTION' || item.tagName === 'OPTGROUP') {
+        const clonedItem = item.cloneNode(true) as HTMLOptionElement | HTMLOptGroupElement;
+        if (clonedItem.tagName === 'OPTION' && (clonedItem as HTMLOptionElement).value === this.value) {
+          (clonedItem as HTMLOptionElement).selected = true;
+        }
+        this.selectEl.appendChild(clonedItem);
+      }
+    });
+
+    this.updateSelectedOption();
+  };
+
+  render() {
+    return (
+      <Host aria-disabled={this.disabled ? 'true' : null}>
+        <div class="pds-select">
+          <PdsLabel htmlFor={this.componentId} text={this.label} />
+          <select
+            autocomplete={this.autocomplete}
+            class="pds-select__field"
+            disabled={this.disabled}
+            id={this.componentId}
+            multiple={this.multiple}
+            name={this.name}
+            onChange={this.onSelectUpdate}
+            required={this.required}
+            ref={(el) => (this.selectEl = el as HTMLSelectElement)}
+          ></select>
+          <div aria-hidden="true" class="hidden" ref={(el) => (this.slotContainer = el)}>
+            <slot onSlotchange={this.handleSlotChange}></slot>
+          </div>
+          {(this.helperMessage || this.errorMessage) && (
+            <div class="pds-select__message">
+              {this.helperMessage && (
+                <p class="pds-select__helper-message" id={messageId(this.componentId, 'helper')}>
+                  {this.helperMessage}
+                </p>
+              )}
+              {this.errorMessage && (
+                <p class="pds-select__error-message" id={messageId(this.componentId, 'error')} aria-live="assertive">
+                  <pds-icon name="danger" size="small" />
+                  {this.errorMessage}
+                </p>
+              )}
+            </div>
+          )}
```

Comment:
Can we clean this up?

Few options
1. Extract it out to a helper method `renderMessages()`
2. Would having an empty div cause problems? If not, then we could simply remove the double checks. L187 and (L189 or L194)

Reply:
Gave this a shot, let me know.